src/Microsoft.ML.Transforms/Text/LdaTransform.cs

```diff
@@ -460,516 +886,228 @@ private static int GetFrequency(double value)
         }
 
-        protected override ColumnType GetColumnTypeCore(int iinfo)
+        private readonly IHost _host;
+        private readonly LdaTransform.ColumnInfo[] _columns;
+
+        /// <include file='doc.xml' path='doc/members/member[@name="LightLDA"]/*' />
+        /// <param name="env">The environment.</param>
+        /// <param name="inputColumn">The column containing text to tokenize.</param>
+        /// <param name="outputColumn">The column containing output tokens. Null means <paramref name="inputColumn"/> is replaced.</param>
+        /// <param name="numTopic">The number of topics in the LDA.</param>
+        /// <param name="alphaSum">Dirichlet prior on document-topic vectors.</param>
+        /// <param name="beta">Dirichlet prior on vocab-topic vectors.</param>
+        /// <param name="mhstep">Number of Metropolis Hasting step.</param>
+        /// <param name="numIterations">Number of iterations.</param>
+        /// <param name="likelihoodInterval">Compute log likelihood over local dataset on this iteration interval.</param>
+        /// <param name="numThreads">The number of training threads. Default value depends on number of logical processors.</param>
+        /// <param name="numMaxDocToken">The threshold of maximum count of tokens per doc.</param>
+        /// <param name="numSummaryTermPerTopic">The number of words to summarize the topic.</param>
+        /// <param name="numBurninIterations">The number of burn-in iterations.</param>
+        /// <param name="resetRandomGenerator">Reset the random number generator for each document.</param>
+        public LdaEstimator(IHostEnvironment env,
+            string inputColumn,
+            string outputColumn = null,
+            int numTopic = Defaults.NumTopic,
+            Single alphaSum = Defaults.AlphaSum,
+            Single beta = Defaults.Beta,
+            int mhstep = Defaults.Mhstep,
+            int numIterations = Defaults.NumIterations,
+            int likelihoodInterval = Defaults.LikelihoodInterval,
+            int numThreads = Defaults.NumThreads,
+            int numMaxDocToken = Defaults.NumMaxDocToken,
+            int numSummaryTermPerTopic = Defaults.NumSummaryTermPerTopic,
+            int numBurninIterations = Defaults.NumBurninIterations,
+            bool resetRandomGenerator = Defaults.ResetRandomGenerator)
+            : this(env, new[] { new LdaTransform.ColumnInfo(inputColumn, outputColumn ?? inputColumn,
+                numTopic, alphaSum, beta, mhstep, numIterations, likelihoodInterval, numThreads, numMaxDocToken,
+                numSummaryTermPerTopic, numBurninIterations, resetRandomGenerator) })
+        { }
+
+        /// <include file='doc.xml' path='doc/members/member[@name="LightLDA"]/*' />
+        /// <param name="env">The environment.</param>
+        /// <param name="columns">Pairs of columns to compute LDA.</param>
+        public LdaEstimator(IHostEnvironment env, LdaTransform.ColumnInfo[] columns)
```

Comment:
> LdaTransform [](start = 49, length = 13)

put params in front.

Reply:
fixed

---
In reply to: [232104418](https://github.com/dotnet/machinelearning/pull/1410#discussion_r232104418) [](ancestors = 232104418)